install.sh

```diff
@@ -254,74 +243,78 @@ function update_device_json () {
 
   if [[ $(jq --arg key "$1" -e '.installed_packages[] | select(.name == $key )' device.json) ]]; then
     echo_intra "Updating version number of ${1} in device.json..."
-    cat <<< $(jq --arg key0 "$1" --arg value0 "$2" '(.installed_packages[] | select(.name == $key0) | .version) |= $value0' device.json) > device.json
+    cp -f device.json device.json.tmp
+    cat <<< "$(jq --arg key0 "$1" --arg value0 "$2" '(.installed_packages[] | select(.name == $key0) | .version) |= $value0' device.json.tmp)" > device.json
   else
     echo_intra "Adding new information on ${1} to device.json..."
-    cat <<< $(jq --arg key0 "$1" --arg value0 "$2" '.installed_packages |= . + [{"name": $key0, "version": $value0}]' device.json ) > device.json
+    cp -f device.json device.json.tmp
+    cat <<< "$(jq --arg key0 "$1" --arg value0 "$2" '.installed_packages |= . + [{"name": $key0, "version": $value0}]' device.json.tmp)" > device.json
   fi
 }
+
 echo_info "Downloading Bootstrap packages...\n"
 # Extract, install and register packages.
 for i in $(seq 0 $((${#urls[@]} - 1))); do
   url="${urls["${i}"]}"
   sha256="${sha256s["${i}"]}"
-  tarfile="$(basename ${url})"
+  tarfile="$(basename "${url}")"
   name="${tarfile%%-*}"   # extract string before first '-'
   rest="${tarfile#*-}"    # extract string after first '-'
-  version="${rest%%-chromeos*}"
-                          # extract string between first '-' and "-chromeos"
+  version="${rest%%-chromeos*}" # extract string between first '-' and "-chromeos"
 
   download_check "${name}" "${url}" "${tarfile}" "${sha256}"
   extract_install "${name}" "${tarfile}"
   update_device_json "${name}" "${version}"
 done
 
 echo_info "\nCreating symlink to 'crew' in ${CREW_PREFIX}/bin/"
-echo -e "${GRAY}"
+
+# Work around https://github.com/chromebrew/chromebrew/issues/3305.
+sudo ldconfig &> /dev/null || true
+echo_out "\nCreating symlink to 'crew' in ${CREW_PREFIX}/bin/"
 ln -sfv "../lib/crew/bin/crew" "${CREW_PREFIX}/bin/"
-echo -e "${RESET}"
 
-echo_info "Setup and synchronize local package repo..."
-echo -e "${GRAY}"
+echo_out "Set up and synchronize local package repo..."
 
 # Remove old git config directories if they exist.
 find "${CREW_LIB_PATH}" -mindepth 1 -delete
 
-curl -L --progress-bar https://github.com/"${OWNER}"/"${REPO}"/tarball/"${BRANCH}" | tar -xz --strip-components=1 -C "${CREW_LIB_PATH}"
+curl -L -O --progress-bar https://github.com/"${OWNER}"/"${REPO}"/tarball/"${BRANCH}"
```

Comment:
Why don't we use pipe? It should be slightly faster without reading and writing on the filesystem

Reply:
It doesn't work :p. Not sure why, but the installs started breaking with the previous setup. I looked into it, but couldn't find an answer. If someone wants to find out the issue and fix it, be my guest, but this way seems like a reasonable tradeoff for me.

Reply:
Running the command separately works for me:
```
curl -L --progress-bar https://github.com/chromebrew/chromebrew/tarball/master | tar -xz --strip-components=1 -C "/usr/local/lib/crew"
```

Reply:
idk :p

But I encountered issues in all 3 of my test environments-- host, chroot and docker.

Idk if its just specific to me, but even if it is, I can't test without the fix.